packages/firestore/test/integration/api/database.test.ts

```diff
@@ -1447,117 +1473,112 @@ apiDescribe('Database', (persistence: boolean) => {
-        expect(doc.get('title')).to.equal('olive');
-        expect(doc.get('author')).to.equal('author');
+        const docSnap = await getDoc(ref);
+        expect(docSnap.get('title')).to.equal('olive');
+        expect(docSnap.get('author')).to.equal('author');
       });
     });
 
     it('DocumentReference.set() supports partials with merge', async () => {
       return withTestDb(persistence, async db => {
-        const ref = db
-          .collection('posts')
-          .doc()
-          .withConverter(postConverterMerge);
-        await ref.set(new Post('walnut', 'author'));
-        await ref.set({ title: 'olive' }, { merge: true });
-        const doc = await ref.get();
-        expect(doc.get('title')).to.equal('olive');
-        expect(doc.get('author')).to.equal('author');
+        const ref = doc(collection(db, 'posts')).withConverter(
+          postConverterMerge
+        );
+        await setDoc(ref, new Post('walnut', 'author'));
+        await setDoc(ref, { title: 'olive' }, { merge: true });
+        const docSnap = await getDoc(ref);
+        expect(docSnap.get('title')).to.equal('olive');
+        expect(docSnap.get('author')).to.equal('author');
       });
     });
 
     it('DocumentReference.set() supports partials with mergeFields', async () => {
       return withTestDb(persistence, async db => {
-        const ref = db
-          .collection('posts')
-          .doc()
-          .withConverter(postConverterMerge);
-        await ref.set(new Post('walnut', 'author'));
-        await ref.set(
+        const ref = doc(collection(db, 'posts')).withConverter(
+          postConverterMerge
+        );
+        await setDoc(ref, new Post('walnut', 'author'));
+        await setDoc(
+          ref,
           { title: 'olive', author: 'writer' },
           { mergeFields: ['title'] }
         );
-        const doc = await ref.get();
-        expect(doc.get('title')).to.equal('olive');
-        expect(doc.get('author')).to.equal('author');
+        const docSnap = await getDoc(ref);
+        expect(docSnap.get('title')).to.equal('olive');
+        expect(docSnap.get('author')).to.equal('author');
       });
     });
 
     it('calls DocumentSnapshot.data() with specified SnapshotOptions', () => {
       return withTestDb(persistence, async db => {
-        const docRef = db.doc('some/doc').withConverter({
-          toFirestore(post: Post): firestore.DocumentData {
+        const docRef = doc(db, 'some/doc').withConverter({
+          toFirestore(post: Post): DocumentData {
             return { title: post.title, author: post.author };
           },
-          fromFirestore(
-            snapshot: firestore.QueryDocumentSnapshot,
-            options: firestore.SnapshotOptions
-          ): Post {
+          fromFirestore(snapshot: QueryDocumentSnapshot): Post {
+            // Hack: Due to our build setup, TypeScript thinks this is a
+            // firestore-lite converter which does not have options
```

Comment:
wondering why I didn't see this hack elsewhere in other tests.

Reply:
None of the other converters use the `SnapshotOptions`, which is the second argument here.